src/Couchbase.Lite.Shared/API/Query/IGroupByRouter.cs

```diff
@@ -34,7 +34,8 @@ public interface IGroupByRouter
         /// <param name="expressions">The clauses to group by</param>
         /// <returns>The query grouped by the given clauses for further processing</returns>
         [NotNull]
-        IGroupBy GroupBy(params IExpression[] expressions);
+        [ContractAnnotation("null => halt")]
+        IGroupBy GroupBy([NotNull]params IExpression[] expressions);
```

Comment:
This is an interesting one.  I don't think `params` can ever be null anyway.  I think what this one needs is `ItemNotNull` so that it contains no null items.  The `Contract` annotation here also is probably not useful.